File: Core/src/com/forcex/anim/AnimationControl.java
package com.forcex.anim;

public class AnimationControl
{
	boolean loop = false;
	boolean play = true;
	
	public static final byte CMD_RESET = 0;
	public static final byte CMD_PLAY = 1;
	public static final byte CMD_PAUSE = 2;
	public static final byte CMD_LOOP = 3;
	public static final byte CMD_NO_LOOP = 4;
	
	public float time = 0.0f;
	public float speed = 1.0f;
	
	public void putCommand(byte cmd){
		switch(cmd){
			case CMD_RESET:
				time = 0.0f;
				speed = 1.0f;
				play = true;
				loop = false;
				break;
			case CMD_PLAY:
				play = true;
				break;
			case CMD_PAUSE:
				play = false;
				break;
			case CMD_LOOP:
				loop = true;
				break;
			case CMD_NO_LOOP:
				loop = false;
				break;
		}
	}
	
	public boolean isRunning(){
		return play;
	}
}

File: Backend/android-backend/app/src/main/java/com/forcex/anim/AnimationControl.kt
package com.forcex.anim

class AnimationControl {
    internal var loop = false
    var isRunning = true
        internal set

    var time = 0.0f
    var speed = 1.0f

    fun putCommand(cmd: Byte) {
        when (cmd) {
            CMD_RESET -> {
                time = 0.0f
                speed = 1.0f
                isRunning = true
                loop = false
            }
            CMD_PLAY -> isRunning = true
            CMD_PAUSE -> isRunning = false
            CMD_LOOP -> loop = true
            CMD_NO_LOOP -> loop = false
        }
    }

    companion object {

        val CMD_RESET: Byte = 0
        val CMD_PLAY: Byte = 1
        val CMD_PAUSE: Byte = 2
        val CMD_LOOP: Byte = 3
        val CMD_NO_LOOP: Byte = 4
    }
}
